src/login/pam_systemd.c

```diff
@@ -211,26 +218,45 @@ static int export_legacy_dbus_address(
         return r;
 }
 
+static int append_session_cg_limit(sd_bus_message *m, const char *limit, const char *name) {
+        uint64_t val;
+        int r;
+
+        if (!isempty(limit)) {
```

Comment:
(not that it matters, but I'd always minimize indentation, and prefer early exit over long if blocks. Hence, my recommendation: just do

```c
if (isempty(limit))
       return 0;
```